src/components/Filters/index.tsx

```diff
@@ -66,482 +30,293 @@ const FilterWrapper = styled(Box)`
 	}
 
-	public getCleanEditModel(field?: string | null) {
-		const schema = this.state.schema;
-		if (!field) {
-			field = Object.keys(schema.properties!).shift()!;
-		}
-
-		const fieldOperators = this.getOperators(field);
-		if (!fieldOperators.length) {
-			return {
-				field,
-				operator: '',
-				value: '',
-			};
-		}
-
-		const operator = fieldOperators.shift()!.slug;
-
-		let value: any = '';
-		const subschema = schema.properties![field];
-		if (typeof subschema !== 'boolean') {
-			if (subschema.enum) {
-				value = subschema.enum[0] || '';
-			}
-
-			if (subschema.oneOf) {
-				value = (subschema.oneOf[0] as JSONSchema6).const || '';
-			}
+	return includes(allowedModes, mode);
+};
 
-			if (subschema.type === 'boolean') {
-				value = true;
-			}
-		}
+const useDeepCompare = (value: any) => {
+	const ref = React.useRef();
 
-		return {
-			field,
-			operator,
-			value,
-		};
+	if (!isEqual(value, ref.current)) {
+		ref.current = value;
 	}
 
-	public getOperators(field: string) {
-		const schema = this.state.schema;
-		return SchemaSieve.getOperators(schema, field);
-	}
+	return ref.current;
+};
 
-	public setEditField(field: string, index: number) {
-		const currentEdit = this.state.edit.slice();
-		currentEdit.splice(index, 1, this.getCleanEditModel(field));
-		this.setState({
-			edit: currentEdit,
-		});
-	}
+const Filters = ({
+	schema,
+	disabled,
+	viewScopes,
+	onFiltersUpdate,
+	onViewsUpdate,
+	addFilterButtonProps,
+	viewsMenuButtonProps,
+	renderMode,
+	dark,
+	compact,
+	...restProps
+}: FiltersProps) => {
+	// Since default values does not work by passing null as a prop value,
+	// I declare them here like so to avoid many checks
+	const filters = restProps.filters ? restProps.filters : [];
+	const views = restProps.views ? restProps.views : [];
+
+	const flatSchema = SchemaSieve.flattenSchema(schema);
+	const flatViews = flattenViews(views);
+	const flatFilters = filters.map(filter => SchemaSieve.flattenSchema(filter));
+	const shouldHideLabel = useBreakpoint(compact || [false]);
+
+	const [filtersState, setFiltersState] = React.useState<FiltersState>({
+		showModal: false,
+		searchString: '',
+		editingFilter: null,
+		filters: flatFilters,
+		views: flatViews,
+		schema: flatSchema,
+		edit: [],
+	});
+
+	React.useEffect(() => {
+		filtersState.edit.push(SchemaSieve.getCleanEditModel(filtersState.schema));
+	}, []);
+
+	React.useEffect(() => {
+		setFiltersState(prevState => ({
+			...prevState,
+			schema: SchemaSieve.flattenSchema(schema),
+		}));
+	}, [schema]);
 
-	public setEditOperator(operator: string, index: number) {
-		const currentEdit = this.state.edit.slice();
-		const item = currentEdit[index];
-		currentEdit.splice(index, 1, { ...item, operator });
-		this.setState({
-			edit: currentEdit,
-		});
-	}
+	React.useEffect(() => {
+		setFiltersState(prevState => ({
+			...prevState,
+			filters: filters.map(filter => SchemaSieve.flattenSchema(filter)),
+		}));
+	}, [useDeepCompare(filters)]);
```

Comment:
Now that I look at the code, why do we even need this effect? We already have all the CRUD methods for filters and views that set the state, so these hooks feel unnecessary?